### PR TITLE
Improve coverage for airflow.security.kerberos module

### PR DESCRIPTION
**Pull request by mik-laj**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/security/test_kerberos.py [hunk 1]
```diff
@@ -104,3 +105,255 @@ def test_args_from_cli(self):
...
+                    stderr=mock_subprocess.PIPE,
+                    stdout=mock_subprocess.PIPE,
+                    universal_newlines=True,
+                ),
+                mock.call.Popen().__enter__(),
+                mock.call.Popen().__enter__().wait(),
+                mock.call.Popen().__exit__(None, None, None),
+                mock.call.call(['kinit', '-c', '/tmp/airflow_krb5_ccache', '-R'], close_fds=True),
+            ]
+        )
```
Comment (uranusjr) [inline tests/security/test_kerberos.py]: `assert_has_calls` [is non-exhaustive](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls). Are we expecting `subprocess.Popen` to be called by anything else here? If not, we should use `assert mock_subprocess.mock_calls == [...]` isntead.

File: tests/security/test_kerberos.py [hunk 2]
```diff
@@ -104,3 +105,255 @@ def test_args_from_cli(self):
                 )
 
             assert ctx.value.code == 1
+
+
+class TestKerberosUnit(unittest.TestCase):
```
Comment (uranusjr) [inline tests/security/test_kerberos.py:109]: Should we use a pytest setup instead? `pytest.mark.paramtrize`, `caplog`, and `pytest.raises` are easier to use than `parameterized`, `assertLogs`, and `assertRaises` IMO. But this is minor and I’m OK if you feel the current approach is more familiar etc.

**Review by uranusjr — COMMENTED** [review]
A couple of minor comments. I don’t understand Kerberos but none the changes seem related to the actual Kerberos functionality.

File: tests/security/test_kerberos.py (hunk 1 above)
Comment (mik-laj) [reply tests/security/test_kerberos.py]: Updated. Thanks. I was concerned if mock.ANY would work in this situation, but it works.

File: tests/security/test_kerberos.py (hunk 2 above)
Comment (mik-laj) [reply tests/security/test_kerberos.py:109]: This is very subjective, but I checked the popularity of each approach and I can see that it is very close.

PS. We had discussions on the mailing list about this. https://lists.apache.org/thread.html/1e4df7d4b0cd9b2d2bb76a3336471aa85e852545dd41ada6d4e461b8%40%3Cdev.airflow.apache.org%3E Feel free to join it

**Comment by potiuk** [comment]
Reopened to rebuild.

**Review by potiuk — APPROVED** [review]

**Pull request merged**